Rubberduck.CodeAnalysis/Inspections/Abstract/InspectionBase.cs

```diff
@@ -86,28 +86,22 @@ protected virtual IEnumerable<Declaration> BuiltInDeclarations
 
         protected bool IsIgnoringInspectionResultFor(QualifiedModuleName module, int line)
         {
-            var annotations = State.GetModuleAnnotations(module).ToList();
-
-            if (State.GetModuleAnnotations(module) == null)
-            {
-                return false;
-            }
-
-            // VBE 1-based indexing
-            for (var i = line; i >= 1; i--)
+            var lineScopedAnnotations = State.DeclarationFinder.FindAnnotations(module, line);
+            foreach (var ignoreAnnotation in lineScopedAnnotations.OfType<IgnoreAnnotation>())
             {
-                var annotation = annotations.SingleOrDefault(a => a.QualifiedSelection.Selection.StartLine == i);
-                var ignoreAnnotation = annotation as IgnoreAnnotation;
-                var ignoreModuleAnnotation = annotation as IgnoreModuleAnnotation;
-
-                if (ignoreAnnotation?.InspectionNames.Contains(AnnotationName) == true)
+                if (ignoreAnnotation.InspectionNames.Contains(AnnotationName))
                 {
                     return true;
                 }
+            }
+
+            var moduleDeclaration = State.DeclarationFinder.Members(module)
```

Comment:
Wouldn't it be more efficient to check for module-level annotations first? Worst-case would be a massive module with many members, right?

Reply:
Sorry, I do not understand what you mean. The part above looking for `IgnoreAnnotations` for the line is a dictionary lookup. Where do the members enter, here?